### PR TITLE
add static factory for ReadResult

### PR DESCRIPTION
**Pull request by Mauin**
#10 

Do you think it makes sense to instead of the one `create(...)` method add multiple for each value of ReadState? 

Secondly do you prefer to also use the static factory method internally or still call the constructor?

### CODE REVIEW

File: whorlwind/src/main/java/com/squareup/whorlwind/ReadResult.java [hunk 1]
```diff
@@ -47,6 +47,14 @@
     this.value = value;
   }
 
+  /**
+   * Creates a new instance of ReadResult
+   */
```
Comment (JakeWharton) [inline whorlwind/src/main/java/com/squareup/whorlwind/ReadResult.java]: Delete this documentation. It's not adding any value.

**Comment by JakeWharton** [comment]
> do you prefer to also use the static factory method internally

Yes please. And let's make the constructor `private` in that case.

File: whorlwind/src/main/java/com/squareup/whorlwind/ReadResult.java [hunk 2]
```diff
@@ -39,14 +39,19 @@
    */
   @Nullable public final ByteString value;
 
-  ReadResult(@NonNull ReadState readState, int code, @Nullable CharSequence message,
+  private ReadResult(@NonNull ReadState readState, int code, @Nullable CharSequence message,
       @Nullable ByteString value) {
     this.readState = readState;
```
Comment (mattprecious) [inline whorlwind/src/main/java/com/squareup/whorlwind/ReadResult.java]: Should we add a null check here (or in the creator) since this is now public?

Comment (Mauin) [inline whorlwind/src/main/java/com/squareup/whorlwind/ReadResult.java]: Yeah, makes sense, even though the Annotation should guard against this it is no guarantee.

**Comment by mattprecious** [comment]
lgtm. @JakeWharton?

**Comment by JakeWharton** [comment]
:+1:

**Comment by mattprecious** [comment]
Thanks!

**Pull request merged**